topics/statistics/tutorials/intro-to-ml-with-python/tutorial.md

```diff
@@ -64,543 +92,485 @@ plt.rc('ytick.major', size=8, pad=12)
+| CE072 | -1.713273 | -1.400956 | -1.543058 | -0.930367 | -1.058800 | -0.455020 | -1.302403 | -0.110293 | -0.332380 | -0.232460 | ... | -0.131733 | -0.070336 | 0.821996 | -1.566652 | 0.914053 | -1.707726 | 0.498226 | -1.500588 | 0.361168 | -1.020456 |
+| CE084 | -0.096239 | -0.599251 | -1.499636 | -0.847275 | -1.171365 | -0.952574 | -1.347691 | 0.561542 | -0.335009 | -0.702851 | ... | -0.729461 | 0.135614 | 1.074398 | 0.629679 | -0.691100 | -1.247779 | 0.167965 | -1.525064 | 0.150271 | 0.105746 |
+| CE110 | -0.712374 | -1.081618 | -1.530316 | -1.259747 | -1.109999 | -0.582357 | -1.233085 | 0.008014 | -0.915632 | -0.746339 | ... | -0.054882 | 0.363344 | 0.720155 | 0.465315 | 1.450199 | -1.706606 | 0.602451 | -1.507727 | -2.207455 | -0.139036 |
 
-    df_nuisance["N"+str(i)]=df_nuisance.N**i
-    
-    list_co.append( "N"+str(i) )
 
-    ## create the model, without the last 2 points
-    model = smf.ols("Breeding ~ " + "+".join(list_co) , data = df_nuisance.iloc[:-2, :] )
-    results = model.fit()#we do the actual fit
-    
-    models.append("+".join(list_co))
-    logLikelihoods.append( results.llf )
-    
-logLikelihoods
+```python
+y = df.loc[i1 , "Flesh Colour"]
+y.describe()
 ```
 
 
 
 
-    [np.float64(-125.86306783486532),
-     np.float64(-112.64380299806213),
-     np.float64(-110.71030133123855),
-     np.float64(-103.39762229283633)]
+    count    73.000000
+    mean     24.473845
+    std      12.437785
+    min       6.992000
+    25%      13.484500
+    50%      24.746500
+    75%      30.996200
+    max      57.035100
+    Name: Flesh Colour, dtype: float64
 
 
 
 
 ```python
-import scipy.stats as stats 
-
-for i in range(1,len(logLikelihoods)):
-    #calculating the pvalue for the LRT between the models
-    pval=1-stats.chi2.cdf(2*(logLikelihoods[i]-logLikelihoods[i-1]),1)
-        
-    print("model {:<10} - model {:<12} : LRT p-value = {:.2e}".format(models[i-1],
-                                                                     models[i],
-                                                                     pval))
-
+sns.histplot(y)
 ```
 
-    model N1         - model N1+N2        : LRT p-value = 2.72e-07
-    model N1+N2      - model N1+N2+N3     : LRT p-value = 4.92e-02
-    model N1+N2+N3   - model N1+N2+N3+N4  : LRT p-value = 1.31e-04
 
 
-You see that the choice between quadratic and cubic is associated to a fairly high p-value here (~0.0492).
 
-Let's check how the model behaves:
+    <Axes: xlabel='Flesh Colour', ylabel='Count'>
 
 
-```python
-from sklearn.metrics import r2_score
-from sklearn.metrics import mean_squared_error
-fig, ax = plt.subplots(1, 2,figsize=(14,7))
-## Quadratic model
 
-for i, variables in enumerate( ["N1+N2","N1+N2+N3"] ):
 
-    model = smf.ols("Breeding ~ "+variables , data = df_nuisance.iloc[:-2] )
-    results = model.fit()#we do the actual fit
     
-    predicted = results.predict( df_nuisance )
+![png](images/outputs/output_8_1.png)
```

Comment:
the text inside the square brackets is the alt text, and should describe the contents of the image for e.g. those using a screen reader. Instead or "PNG" something like "Bar chart showing .." i.e. anything somebody who can not see the image should know about it

Reply:
your suggestion is already better :+1: 

I guess just use your best judgement, the main point is that the tutorial is also understandable without seeing the images, and if for example in the tutorial text you already discuss whats in the image (as you often do), the alt text is less important, etc

Reply:
OK, I'll implement this & the updated LOs when Berenice also turns in her review.